object_storage_api/auth/jwt_middleware.py

```diff
@@ -17,37 +21,39 @@
 
 logger = logging.getLogger()
 
+security = HTTPBearer(auto_error=True)
 
-class JWTBearer(HTTPBearer):
+
+class JWTMiddleware(BaseHTTPMiddleware):
     """
-    Extends the FastAPI `HTTPBearer` class to provide JSON Web Token (JWT) based authentication/authorization.
+    A middleware class to provide JSON Web Token (JWT) based authentication/authorization.
     """
 
-    def __init__(self, auto_error: bool = True) -> None:
-        """
-        Initialize the `JWTBearer`.
-
-        :param auto_error: If `True`, it automatically raises `HTTPException` if the HTTP Bearer token is not provided
-            (in an `Authorization` header).
+    async def dispatch(self, request: Request, call_next: RequestResponseEndpoint) -> Response:
         """
-        super().__init__(auto_error=auto_error)
+        Performs JWT access token authentication/authorization before processing the request.
 
-    async def __call__(self, request: Request) -> str:
-        """
-        Callable method for JWT access token authentication/authorization.
-
-        This method is called when `JWTBearer` is used as a dependency in a FastAPI route. It performs authentication/
-        authorization by calling the parent class method and then verifying the JWT access token.
-        :param request: The FastAPI `Request` object.
+        :param request: The Starlette `Request` object.
+        :param call_next: The next function to call to process the `Request` object.
         :return: The JWT access token if authentication is successful.
         :raises HTTPException: If the supplied JWT access token is invalid or has expired.
         """
-        credentials: HTTPAuthorizationCredentials = await super().__call__(request)
-
-        if not self._is_jwt_access_token_valid(credentials.credentials):
-            raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail="Invalid token or expired token")
-
-        return credentials.credentials
+        if (
+            not request.url.path == f"{config.api.root_path}/docs"
+            and not request.url.path == f"{config.api.root_path}/openapi.json"
+        ):
```

Comment:
You can save a few lines by doing
```suggestion
        if request.url.path not in [f"{config.api.root_path}/docs", f"{config.api.root_path}/openapi.json"]:
```